workflow/templateresolution/context.go

```diff
@@ -166,6 +199,17 @@ func (ctx *Context) GetTemplateScope() string {
 	return string(ctx.tmplBase.GetResourceScope()) + "/" + ctx.tmplBase.GetName()
 }
 
+// UpdateTemplateStatus update the WorkflowTemplate.Status of a given WorkflowTemplate.
+func (ctx *Context) UpdateTemplateStatus(wftmple *wfv1.WorkflowTemplate) (*wfv1.WorkflowTemplate, error) {
+	if ctx.wftmplClient == nil {
+		ctx.log.Warnln("while try update template status, wftmplClient not set")
```

Comment:
This error message is coming out when running the [dag diamond](https://github.com/argoproj/argo-workflows/blob/main/examples/workflow-template/dag.yaml), which probably shouldn't happen. Could you investigate.

Reply:
I believe you run inside a devcontainer. So I assume you're testing with `make start`.

Assuming you are in the argo-workflows checkout directory - the logs will be in `logs/controller.log`. You can run `kubectl create -f examples/dag_diamond.yaml` and I got this error.

Reply:
Thank you, I ran in my machine directly, and got the logs in `logs/controller.log` too.

Reply:
But, I can't reproduce it. Those are what I did.
`
git checkout <target-branch>
make clean
make start UI=true

 kubectl create -f examples/dag-diamond.yaml
    + kubectl create -f examples/dag-diamond.yaml
    workflow.argoproj.io/dag-diamond-67fvs created
    
➜  logs git:(support-record-preset) cat controller.log|grep while
➜  logs git:(support-record-preset)
`
And I checked other files in the `logs`, not found  the error log too.
Did I miss something? or could you recheck it?

Reply:
I tried to reproduce by running dag workflow with a tmpl-ref, with the following commands:
```
kubectl create -f examples/workflow-template/templates.yaml
kubectl create -f examples/workflow-template/dag.yaml
```
but I still couldn't find the error log, so I mark this conversation resolved. If I missed something, feel free to reopen it.
Thank you a lot.